lib/path.h

```diff
@@ -150,22 +151,40 @@ class CPPCHECKLIB Path {
      * @brief Identify language based on file extension.
      * @param path filename to check. path info is optional
      * @return true if extension is meant for C files
+     * @deprecated does not account for headers - use @identify() instead
      */
-    static bool isC(const std::string &path);
+    static DEPRECATED bool isC(const std::string &path);
 
     /**
      * @brief Identify language based on file extension.
      * @param path filename to check. path info is optional
      * @return true if extension is meant for C++ files
+     * @deprecated returns true for some header extensions - use @identify() instead
      */
-    static bool isCPP(const std::string &path);
+    static DEPRECATED bool isCPP(const std::string &path);
 
     /**
      * @brief Is filename a header based on file extension
      * @param path filename to check. path info is optional
      * @return true if filename extension is meant for headers
+     * @deprecated returns only heuristic result - use @identify() or @isHeader2() instead
      */
-    static bool isHeader(const std::string &path);
+    static DEPRECATED bool isHeader(const std::string &path);
+
+    /**
+     * @brief Is filename a header based on file extension
```

Comment:
I believe this description is how `isHeader` is supposed to work.

Reply:
`isHeader()` checks if the file extension starts with `h`. That would be working fine if it processes only files which match the list of supported files but this function I used to for that filtering so it cannot be heuristic only and needs to be exact..

Reply:
Also if somebody uses Cppcheck as a library having any `.h*` file matches as header is definitely wrong. That's the reason I added the `2` suffix so the existing function isn't silently changing it behavior leading to unexpected behavior.